src/main/java/seedu/pill/util/ItemList.java

```diff
@@ -0,0 +1,94 @@
+package seedu.pill.util;
+
+import java.util.ArrayList;
+
+/**
+ * Represents a list of items and provides methods to add, delete, list, and edit items.
+ */
+public class ItemList {
+    private ArrayList<Item> items;
+
+    /**
+     * Constructor for ItemList.
+     * Initializes the internal ArrayList to store items.
+     */
+    public ItemList() {
+        this.items = new ArrayList<>();
+    }
+
+    /**
+     * Adds a new item to the list.
+     *
+     * @param name     The name of the item.
+     * @param quantity The quantity of the item.
+     */
+    public void addItem(String name, int quantity) {
+        if (name == null || name.trim().isEmpty() || quantity <= 0) {
+            System.out.println("Invalid item name or quantity.");
+            return;
+        }
```

Comment:
Should the predicate (condition) here be abstracted out so that its more readable?